src/Strings.h

```diff
@@ -0,0 +1,388 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Details {
+
+template<typename IteratorT>
+inline std::vector<IteratorT> IteratorVectorGenerator(IteratorT const & begin,
+                                                     IteratorT const & end) {
+    std::vector<IteratorT> wordIterVector;
+    wordIterVector.reserve(static_cast<unsigned int>(std::distance(begin, end) + 1));
```

Comment:
> wordIterVector.reserve(static_cast<unsigned int>(std::distance(begin, end) + 1)); [](start = 3, length = 82)

Expensive and unnecessary. reserve only make sense if the length is known in advance. In this case, re-allocation will amortize nicely.